File: backend/VocabuAI.Api/Dtos/FlashCards/FlashCardDto.cs
namespace VocabuAI.Api.Dtos.FlashCards;

// Keep in sync with frontend/src/domain/dtos/flashcards/FlashCardDto.ts

public sealed record FlashCardDto(
    int Id,
    string ForeignLanguage,
    string LocalLanguage,
    string? Synonyms,
    string? Annotation,
    DateTimeOffset DateTimeCreated,
    DateTimeOffset DateTimeUpdated
);

File: frontend/src/domain/dtos/flashcards/FlashCardDto.ts
// Keep in sync with backend/VocabuAI.Api/Dtos/FlashCards/FlashCardDto.cs
export type FlashCardDto = {
  id: number;
  foreignLanguage: string;
  localLanguage: string;
  synonyms?: string | null;
  annotation?: string | null;
  box: number;
  lastAnsweredAt?: string | null;
  dateTimeCreated: string;
  dateTimeUpdated: string;
};
